perfkitbenchmarker/gcp/gce_network.py

```diff
@@ -66,29 +119,16 @@ def AllowPort(self, vm, port):
     with self._lock:
       firewall_name = ('perfkit-firewall-%s-%d' %
                        (FLAGS.run_uri, port))
-      if firewall_name in self.firewall_names:
+      firewall_rule = GceFirewallRule(firewall_name, self.project, port)
+      if firewall_rule in self.firewall_rules:
```

Comment:
I believe this will always return `False` see: https://docs.python.org/2/reference/datamodel.html#object.__hash__

```
User-defined classes have __cmp__() and __hash__() methods by default; with them, all objects compare unequal (except with themselves) and x.__hash__() returns a result derived from id(x).
```